common/djangoapps/student/models.py

```diff
@@ -884,7 +913,8 @@ def is_course_full(self, course):
         """
         is_course_full = False
         if course.max_student_enrollments_allowed is not None:
-            is_course_full = self.num_enrolled_in(course.id) >= course.max_student_enrollments_allowed
+            is_course_full = self.num_enrolled_in_exclude_admins(course.id) >= course.max_student_enrollments_allowed
```

Comment:
1) Yes this will effect both ccx and non ccx course
2) No these tests are only for ccx
3) Currently there are no tests for coursefull or master course (non ccx)